benchmarks/sort/sort_benchmark.cpp

```diff
@@ -90,57 +114,116 @@ struct SortKokkos
     return bin_sort.get_permute_vector();
   }
 
-  static void sort(Kokkos::View<ValueType *, DeviceType> view)
+  static void sort(Kokkos::View<ValueType *, MemorySpace> view)
   {
     auto permute = sortAndComputePermutation(view);
     std::ignore = permute;
   }
 
-  static void applyPermutation(Kokkos::View<SizeType *, DeviceType> permute,
-                               Kokkos::View<ValueType *, DeviceType> in,
-                               Kokkos::View<ValueType *, DeviceType> &out)
+  static void applyPermutation(Kokkos::View<SizeType *, MemorySpace> permute,
+                               Kokkos::View<ValueType *, MemorySpace> in,
+                               Kokkos::View<ValueType *, MemorySpace> &out)
   {
-    using ExecutionSpace = typename DeviceType::execution_space;
     int const n = in.extent(0);
+
     Kokkos::parallel_for(
-        Kokkos::RangePolicy<ExecutionSpace>(0, n),
+        "apply_permutation",
+        Kokkos::RangePolicy<execution_space>(execution_space{}, 0, n),
         KOKKOS_LAMBDA(int const i) { out(permute(i)) = in(i); });
   }
 };
 
-#if defined(KOKKOS_ENABLE_SERIAL)
-template <typename ValueType, typename DeviceType, typename SizeType>
-struct SortStd
+template <typename ValueType, typename ExecutionSpace, typename MemorySpace,
+          typename SizeType>
+struct KokkosHelper<
+    ValueType, ExecutionSpace, MemorySpace, SizeType,
+    std::enable_if_t<!is_accessible_from<MemorySpace, ExecutionSpace>::value>>
 {
   using value_type = ValueType;
-  using device_type = DeviceType;
+  using execution_space = ExecutionSpace;
+  using memory_space = MemorySpace;
 
-  static void check_exec_space(Kokkos::View<ValueType *, DeviceType>)
+  static auto
+  sortAndComputePermutation(Kokkos::View<ValueType *, memory_space> view)
   {
-    static_assert(std::is_same<Kokkos::Serial,
-                               typename DeviceType::execution_space>::value,
-                  "");
+    int const n = view.extent(0);
+
+    auto view_mirror =
+        Kokkos::create_mirror_view_and_copy(execution_space{}, view);
```

Comment:
If we use instances, it would make sense to at least reuse it if possible.

Reply:
Not sure what you mean.